lib/widgets/pokemon_tile_item.dart

```diff
@@ -18,10 +18,7 @@ class PokemonTileItem extends StatelessWidget {
   Widget build(BuildContext context) {
     final pokemonImageUrl = pokemon.pokemon.url.toCustomUrl;
     return GestureDetector(
-      /// TODO: details route will be made static in the pokemon details connector
-      onTap: () => context.pushNamed(PokemonDetailsPage.routeName, extra: pokemon),
-
-      /// TODO: on tap should redirect and display the details of the pokemon
+      onTap: () => context.pushNamed(PokemonDetailsConnector.route, extra: pokemon.pokemon.url.getPokemonId),
```

Comment:
this is the purpose why you created the dto in the first place, for the fact of adding the `id` property in it. Then why resorted to just use the url property then do the string manipulation once again? I was hoping that you fully understood why we decided to create a `dto` since you said to me that you fully understood and got the whole point of it during that time 😓 

also, kindly wrap it in a separate function and name it as `navigateToPokemonDetailsPage`